src/Fantomas.Core/CodePrinter.fs

```diff
@@ -1724,6 +1558,156 @@ let genExpr (e: Expr) =
     | Expr.IndexFromEnd node -> !- "^" +> genExpr node.Expr |> genNode node
     | Expr.Typar node -> genSingleTextNode node
 
+let genChain node =
```

Comment:
I don't see much value in extracting the code from a match clause into a function when that function will only be called from one location (being the match clause body).

Ever since Dallas, I've found it easier to leave that code in the match clause body.

And again, make your PR diff as small as possible ;)

Reply:
Good catch, I had that pulled out when I needed it more than once, forgot to put it back 🙃

Reply:
No worries, I typically revert the entire file in this kind of situation.
`git checkout <hash-on-main> --files src/Fantomas.Core/CodePrinter.fs`